src/shell/shell_cmds.cpp

```diff
@@ -2167,3 +2168,46 @@ void DOS_Shell::CMD_VER(char *args)
 		         dos.version.major, dos.version.minor);
 	}
 }
+
+void DOS_Shell::CMD_VOL(char *args)
+{
+	HELP("VOL");
+
+	uint8_t drive_index;
+	size_t len = 0;
+	if (args) {
+		StripSpaces(args);
+		len = strlen(args);
+	}
+	if (len > 0) {
+		if (len < 2 || args[1] != ':') {
+			SyntaxError();
+			return;
+		}
+		drive_index = toupper(*args) - 'A';
+	} else {
+		drive_index = DOS_GetDefaultDrive();
+	}
+	char drive_letter = drive_index + 'A';
+
+	reg_ah = 0x44;            // DOS IOCTL
+	reg_al = 0x0D;            // Generic block device request
+	reg_bl = drive_index + 1; // Drive (0 = default drive, A = 1, B = 2, etc.)  DOSBox drives start A = 0, so add 1.
+	reg_ch = 0x08;            // Device type (0x08 for block device)
+	reg_cl = 0x66;            // Get volume information (returns serial number, volume label, and filesystem type)
+	reg_dx = 0;               // Offset pointer for return information (DS:DX). Set to 0 for beginning of data segment (hopefully this doesn't step on anyone else's memory?)
```

Comment:
Suggest placing the comments above each variable, then run the commit formatter; it will wrap the comments into blocks.

Reply:
+1
We really dislike end of line comments here, in general 😄

Reply:
(But nice comments otherwise 🎉 )

Reply:
Fixed